### PR TITLE
RavenDB-14114 - Reduce the number of log messages in ReduceMemoryUsage

### PR DESCRIPTION
**Pull request by grisha-kotler**

### CODE REVIEW

File: src/Raven.Server/Documents/Indexes/Index.cs [hunk 1]
```diff
@@ -1418,8 +1415,13 @@ private void ReduceMemoryUsage(StorageEnvironment environment)
                 _currentMaximumAllowedMemory = DefaultMaximumMemoryAllocation;
 
                 var afterFree = NativeMemory.CurrentThreadStats.TotalAllocated;
-                if (_logger.IsInfoEnabled)
-                    _logger.Info($"After cleanup, using {new Size(afterFree, SizeUnit.Bytes)} by '{Name}'.");
+
```
Comment (karmeli87) [inline src/Raven.Server/Documents/Indexes/Index.cs:1418]: Do we want to check the previous `afterFree` with the current `beforeFree`, to skip all of those `Clean` methods?

Comment (ppekrol) [reply src/Raven.Server/Documents/Indexes/Index.cs:1418]: Not sure that I follow? You can get `afterFree` only you execute them.

Comment (karmeli87) [reply src/Raven.Server/Documents/Indexes/Index.cs:1418]: From the previous run

Comment (ppekrol) [reply src/Raven.Server/Documents/Indexes/Index.cs:1418]: cc @grisha-kotler ?

Comment (grisha-kotler) [reply src/Raven.Server/Documents/Indexes/Index.cs:1418]: @karmeli87 good idea.
https://github.com/ravendb/ravendb/pull/9781

**Pull request merged**